nvi-rest/src/main/java/no/sikt/nva/nvi/rest/fetch/FetchNviCandidateByPublicationIdHandler.java

```diff
@@ -48,6 +55,8 @@ protected CandidateDto processInput(Void input, RequestInfo requestInfo, Context
                    .map(identifier -> Candidate.fetchByPublicationId(() -> identifier, candidateRepository,
                                                                      periodRepository))
                    .map(this::checkIfApplicable)
+                   .map(candidate -> validateViewingScope(viewingScopeValidator, RequestUtil.getUsername(requestInfo),
```

Comment:
Is there a reason to not do this _much_ earlier?

Reply:
It would be better to verify before fetching candidate, but we need organizations from NVI candidate to check if user belongs to one of organizations, so it should be done after we have fetched candidate from database.